MdeModulePkg/Library/DxeReportStatusCodeLib/ReportStatusCodeLib.c

```diff
@@ -611,3 +611,80 @@ ReportDebugCodeEnabled (
 {
   return (BOOLEAN)((PcdGet8 (PcdReportStatusCodePropertyMask) & REPORT_STATUS_CODE_PROPERTY_DEBUG_CODE_ENABLED) != 0);
 }
+
+// MU_CHANGE [BEGIN] - Add constructor to cache protocol or register for protocol notification
+
+/**
+  Notification function called when the Status Code Runtime protocol is installed. Locates and caches the protocol.
+
+    @param    Event           Not Used.
+    @param    Context         Not Used.
+
+   @retval   none
+ **/
+VOID
+EFIAPI
+OnStatusCodeRuntimeProtocolInstalledNotification (
+  IN EFI_EVENT  Event,
+  IN VOID       *Context
+  )
+{
+  InternalGetReportStatusCode ();
+
+  gBS->CloseEvent (Event);
+}
+
+/**
+    The constructor attempts to locate and cache the mReportStatusCodeLibStatusCodeProtocol. If not found, it will
+    register a notify event triggered
+
+    @param  ImageHandle   The firmware allocated handle for the EFI image.
+    @param  SystemTable   A pointer to the EFI System Table.
+
+    @retval EFI_SUCCESS   The constructor always returns EFI_SUCCESS.
+
+**/
+EFI_STATUS
+EFIAPI
+DxeReportStatusCodeLibConstructor (
+  IN EFI_HANDLE        ImageHandle,
+  IN EFI_SYSTEM_TABLE  *SystemTable
+  )
+{
+  EFI_EVENT   Event;
+  EFI_STATUS  Status;
+  VOID        *Registration;
+
+  InternalGetReportStatusCode ();
+
+  // If the Status Code Runtime Protocol is not found, register a notification event for when it is installed.
+  if ((mReportStatusCodeLibStatusCodeProtocol == NULL) && (gBS != NULL)) {
+    Status = gBS->CreateEvent (
+                    EVT_NOTIFY_SIGNAL,
+                    TPL_CALLBACK,

```

Comment:
nit: we probably can make this run at a higher level. But this is just to extend the coverage of RSC calls.